src/main/java/io/swagger/service/TransactionService.java

```diff
@@ -1,16 +1,73 @@
 package io.swagger.service;
 
-import io.swagger.repository.AccountRepository;
+import io.swagger.exception.BadInputException;
+import io.swagger.exception.LimitReachedException;
+import io.swagger.exception.NotFoundException;
+import io.swagger.model.Account;
+import io.swagger.model.Transaction;
 import io.swagger.repository.TransactionRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.threeten.bp.OffsetDateTime;
+
+import java.util.List;
+import java.util.Optional;
 
 @Service
 public class TransactionService {
     private TransactionRepository transactionRepository;
+    private AccountService accountService;
+    private UserService userService;
 
-    @Autowired
-    public TransactionService(TransactionRepository transactionRepository){
+    public TransactionService(TransactionRepository transactionRepository, AccountService accountService, UserService userService) {
         this.transactionRepository = transactionRepository;
+        this.accountService = accountService;
+        this.userService = userService;
+    }
+
+    public void addTransaction(Transaction transaction) throws BadInputException, NotFoundException, LimitReachedException {
```

Comment:
In deze methode gebruik je meerderre queries die allemaal níét fout mogen gaan. Als de eerste wel goed gaat en de tweede bijvoorbeeld niet, dan is er bij het eerste account al geld vanaf en bij het tweede account geen geld bij, aka een groot probleem. Als het goed is kun je boven deze methode '@Transactional' zetten (heeft toevallig niks te maken met de transaction die wij gebruiken). Deze annotatie zorgt er als het goed is voor dat alle queries eerst uitgeprobeerd worden en als er iets fout gaat dan worden ze allemaal NIET uitgevoerd. Als ze allemaal goed gaan gaan worden ze vervolgens uitgevoerd